build/db.mk

```diff
@@ -0,0 +1,117 @@
+# ------------------------------------------------------------
+# Copyright 2023 The Radius Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#    
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------
+
+POSTGRES_USER=postgres
+POSTGRES_PASSWORD=ucprulz!
+POSTGRES_IMAGE=ghcr.io/radius-project/mirror/postgres:latest
+POSTGRES_CONTAINER_NAME=radius-postgres
+
+##@ Database
+
+.PHONY: db-init
+db-init: db-dependency-docker-running ## Initialize a local PostgresSQL database for testing
+	@echo "$(ARROW) Initializing local PostgresSQL database"
+	@if [ "$$( docker container inspect -f '{{.State.Running}}' $(POSTGRES_CONTAINER_NAME) 2> /dev/null)" = "true" ]; then \
+		echo "PostgresSQL container $(POSTGRES_CONTAINER_NAME) is already runnning"; \
+	elif [ "$$( docker container inspect -f '{{.State.Running}}' $(POSTGRES_CONTAINER_NAME) 2> /dev/null)" = "false" ]; then \
+		echo "PostgresSQL container $(POSTGRES_CONTAINER_NAME) is not running"; \
+		echo "This might have been a crash"; \
+		echo ""; \
+		docker logs $(POSTGRES_CONTAINER_NAME); \
+		echo ""; \
+		echo "Restarting PostgresSQL container $(POSTGRES_CONTAINER_NAME)"  \
+		docker start $(POSTGRES_CONTAINER_NAME) 1> /dev/null; \
+	else \
+		docker run \
+			--detach \
+			--name $(POSTGRES_CONTAINER_NAME) \
+			--publish 5432:5432 \
+			--env POSTGRES_USER=$(POSTGRES_USER) \
```

Comment:
could this command fail? what if 5432 on host is not available?

Reply:
Yes. It if fails it fails :) Do you think we should try to do something about it?